build_web_compilers/lib/src/dart2wasm_bootstrap.dart

```diff
@@ -17,20 +17,28 @@ import 'web_entrypoint_builder.dart';
 
 final _resourcePool = Pool(maxWorkersPerTask);
 
+/// Result of invoking the `dart2wasm` compiler.
+///
+/// The `supportExpression` is an optional JavaScript expressions emitted by
+/// newer versions of the compiler. It can be evaluated as part of a loader
+/// script to determine whether the current JavaScript environment supports
+/// `dart2wasm`.
+typedef Dart2WasmBootstrapResult = ({String? supportExpression});
```

Comment:
Please make it a class rather than a typedef+record, it's more maintainable. Once `supportExpression` is a field the docs for it can go on the field, which will help, it's a lot to explain in one comment :)

I suggest returning non-null `Dart2WasmBootstrapResult` everywhere and giving it a boolean to say whether it succeeded, rather than using null for failure.